backend/internal/db/dbgorm/deployments.go

```diff
@@ -9,12 +9,17 @@ import (
 	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
 	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
 	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+	"gorm.io/gorm"
 )
 
+func (db *gormdb) createDeployment(ctx context.Context, deployment scheme.Deployment) error {
```

Comment:
hmm.. maybe you are right. I just kept the original logic/signature for returning objects by value.
let's think together? will it be better for us/performance? Or it will change nothing? If it changes nothing, I think I will prefer to keep this by value, then